lib/index.js

```diff
@@ -225,10 +224,6 @@ api.set = async ({
 
   let tokenTypeValue;
   if(type === 'nonce') {
-    // include old value in the query to ensure that no concurrent
-    // changes occurred via another process
-    const oldValue = tokens.length > 0 ? tokens.slice() : {$in: [null, []]};
-    query['meta.bedrock-authn-token.tokens.nonce'] = oldValue;
```

Comment:
Why was this removed? My understanding is that, previously, if another process added a new token or cleared one that was used up that this code would prevent those changes from being dropped. That's probably why `filterExpiredTokens` was also set to `false` ... we may need a different fix here.

Reply:
Hmm well if we set `filterExpiredTokens` this code does not work. Any suggestions on what to do here then if we need to keep this?

Reply:
`oldValue` is this case is looking at the filtered tokens. I assume we need to get the unfiltered tokens to set is the oldValue so it can query correctly?

Reply:
Yes, what this comes down to is that we can't use that `filterExpiredTokens` flag (we can't set it to `true`). We'll have to do the expiration checking ourselves or change something else in the API to enable us to both get the old token value and have it return to use just the non-expired tokens. Perhaps when `filterExpiredTokens` is true we could all of the tokens, the filtered tokens, and the expired tokens in three different properties (as an example). We'll want some kind of elegant solution here that allows us to avoid repeating code.

Reply:
I changed the getAll function to return an object instead:
`
{
    allTokens,
    filteredTokens,
    expiredTokens
}
`

There is currently no use for `expiredTokens` at the moment, so should we return that data just incase we need it for future use?

Reply:
Yes, let's just return it -- and perhaps call `filteredTokens` something like `unexpiredTokens`? (or a better name if you have one, but filtered is pretty non-descript).